lib/lightning/kafka_triggers.ex

```diff
@@ -191,4 +194,57 @@ defmodule Lightning.KafkaTriggers do
 
     %{interval: 10_000, messages_per_interval: messages_per_interval}
   end
+
+  def notify_users_of_trigger_failure(trigger_id) do
+    sending_at = DateTime.utc_now()
+
+    if send_notification?(sending_at, last_notification_sent_at(trigger_id)) do
+      notify_users(trigger_id)
+
+      track_notification_sent(trigger_id, sending_at)
+
+      notify_any_other_nodes(trigger_id, sending_at)
+    end
+  end
+
+  defp last_notification_sent_at(trigger_id) do
+    :persistent_term.get(failure_notification_tracking_key(trigger_id), nil)
+  end
```

Comment:
Persistent terms are localised to a single node, how are we guaranteeing that the calls to this function originate from the same process?

Reply:
@stuartc We are not. This means that there is a chance that the Pipeline on another node also wants to send a notification before it has received the [event](https://github.com/OpenFn/lightning/pull/2568/files#diff-5277a74a56ffb8714c3f4d6ae7afed0781ff27eef37ada7cdd8e606d71392363R34) notifying it that a notification has been sent.

In the worst case, this would mean that there would be a duplicate message for each node in the cluster. However, this risk exists already - at a high enough message rate, it is conceivable that each node experiences such an error simultaneously and they all send the notifications. 

However, once the event is received by all nodes, the window for such duplicated notifications will close. If a node never receives the event, then it will be vulnerable to sending a duplicate notification, but then it will also update the local :persistent_term and will not send any further.

Given that the probability of a persistence error is currently hard to quantify, I elected to bias towards a simpler process that has the risk of sending a small number of duplicate messages rather than sending none at all. In fact, this can be considered the principle for all the code I am going to want to merge in for persistence error handling: 'Messy, but hopefully robust' :).

Note: I have marked this PR as draft for now as I am awaiting feedback on the email contents, but please feel free to add further comments.

Reply:
Ok cool, yeah I agree generally with your approach - we _do_ have the rate limiter used to debounce emails. So that is also an option - but since this is an edge case and has a limited impact I'm comfortable with this.

Reply:
@stuartc Ah, I was not aware that there was existing functionality. Do you mean [this](https://github.com/OpenFn/lightning/blob/main/lib/lightning/pipeline/failure_alerter.ex#L36)?

Reply:
Yes thats it, well more specifically wrapping calls in `Hammer.check_rate` - under the hood it uses mnesia to keep counts in buckets.

Reply:
Ah, cool. I did originally consider Hammer but it seemed overkill for what I needed. Given that we already have the plumbing in place, I can take another look.

Happy for me to do so in a follow-up PR?